controllers/gcpmachine_controller.go

```diff
@@ -301,6 +308,28 @@ func (r *GCPMachineReconciler) getOrCreate(scope *scope.MachineScope, computeSvc
 	return instance, nil
 }
 
+func (r *GCPMachineReconciler) getAddresses(computeSvc *compute.Service, instance *gcompute.Instance) ([]v1.NodeAddress, error) {
+	addresses := []v1.NodeAddress{}
+	for _, nic := range instance.NetworkInterfaces {
+		internalAddress := v1.NodeAddress{
+			Type:    v1.NodeInternalIP,
+			Address: nic.NetworkIP,
+		}
+		addresses = append(addresses, internalAddress)
+
+		// If access configs are associated with this nic, dig out the external IP
```

Comment:
Some context around the access configs that I removed from the comments:

// Note: this follows the logic in debug output from gcloud cli where
// we can just take the first access config and pull out the natIP. Example cli output below:
//
//  INFO: Display format: "    table(
// 		...
// 		networkInterfaces[].networkIP.notnull().list():label=INTERNAL_IP,
// 		networkInterfaces[].accessConfigs[0].natIP.notnull().list()      :label=EXTERNAL_IP,
// 		...
// 	  )"
//
//   NAME                  ZONE           MACHINE_TYPE   PREEMPTIBLE  INTERNAL_IP  EXTERNAL_IP     STATUS
//   test1-controlplane-0  us-central1-a  n1-standard-2               10.128.0.41  xxx.xxx.xxx.xxx  RUNNING